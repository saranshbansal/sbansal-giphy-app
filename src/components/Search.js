import React, { Component } from "react";
import PropTypes from "prop-types";

class Search extends Component {
  static propTypes = {
    onSearch: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleKeyUp = this.handleKeyUp.bind(this);
  }

  state = { value: "" };

  handleChange(event) {
    const { value } = event.target;

    this.setState({
      value
    });

    if (value.length > 2) {
      this.props.onSearch(value);
    }
  }

  handleKeyUp(event) {
    if (event.keyCode === 13) {
      this.props.onSearch(this.state.value);
    }
  }

  render() {
    return (
      <div className="search-box">
        <input
          name="search-gif-input"
          type="text"
          onChange={this.handleChange}
          onKeyUp={this.handleKeyUp}
          placeholder="Search..."
          value={this.state.value}
          aria-label="Search Keyword"
          aria-required="true"
        />
      </div>
    );
  }
}

export default Search;
