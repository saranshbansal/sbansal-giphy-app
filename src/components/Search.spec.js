import React from "react";
import { shallow, mount } from "enzyme";
import Search from "./Search";

describe("Search", () => {
  const onSearchMock = jest.fn();

  it("should render correctly.", () => {
    const wrapper = shallow(<Search onSearch={onSearchMock} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should call onChange prop with input value", () => {
    const event = {
      preventDefault() {},
      target: { name: "search-gif-input", value: "the-value" }
    };

    const wrapper = mount(<Search onSearch={onSearchMock} />);
    expect(wrapper.state("value")).toBe("");

    const searchInput = wrapper.find("input");

    searchInput.simulate("change", event);

    expect(wrapper.state("value")).toBe(event.target.value);

    // Only call the search fn if the input > 2 letters.
    if (event.target.value.length > 2) expect(onSearchMock).toBeCalledWith(event.target.value);
    else expect(onSearchMock).not.toHaveBeenCalled();
  });
});
