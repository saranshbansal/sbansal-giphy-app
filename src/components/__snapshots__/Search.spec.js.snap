// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly. 1`] = `
<div
  className="search-box"
>
  <input
    name="search-gif-input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;
